packages/core/src/StateNode.ts

```diff
@@ -1151,9 +1156,9 @@ class StateNode<
 
   private resolveTransition(
     stateTransition: StateTransition<TContext, TEvent>,
-    currentState?: State<TContext, TEvent, any, any>,
-    _event: SCXML.Event<TEvent> = initEvent as SCXML.Event<TEvent>,
-    context: TContext = this.machine.context
```

Comment:
this is the only line that has truly changed - the other stuff is just reordering etc

The reason behind the infinite loop was that this code was running within a "service scope" and thus it has been executing the `spawnMachine`. The problem was that this `this.machine.context` is actually a getter (yet one reason to hate getters for me :P) and it was executed almost **always** because the default value for the `context` parameter had to be assigned here. In the vast majority of use cases, we are only interested in `currentState.context` though - and we were chosing that a couple of lines below:
https://github.com/statelyai/xstate/blob/56842cb39110923afe70280dde744bea4f81d6dd/packages/core/src/StateNode.ts#L1174
but that was only happening **after** a default value for `context` was already assigned and the contained side-effect logic executed.

So new child actors were constantly created and their initial actions were constantly executed in response to `sendParent` that was contained in those initial actions - thus creating an infinite loop and a memory leak.

I've fixed this by getting rid of this default value and by just providing the appropriate context explicitly at 2 call sites of this function